src/core/Typography/Text/Text.tsx

```diff
@@ -53,34 +56,38 @@ export type TextProps<
  * <Text isMuted>Some muted text.</Text>
  * <Text isSkeleton>Skeleton text</Text>
  */
-export const Text = <T extends React.ElementType | React.ComponentType = 'div'>(
-  props: TextProps<T>,
-) => {
-  const {
-    variant = 'body',
-    as: Element = 'div',
-    className,
-    isMuted = false,
-    isSkeleton = false,
-    ...rest
-  } = props;
+export const Text: TextComponent = React.forwardRef(
```

Comment:
Why cast to `TextComponent`? Will it still allow to pass ref?

Reply:
Read PR description 😄 and yes, TextProps includes ref